### DISCUSSION TITLE
Add example of filtering

### DISCUSSION THREAD

**Discussion by AzHicham** [apalis-dev/apalis#152] [Q&A] [answered]
Hello,

I didn’t find the way to filter process able job based on a predicate.
My goal is to run two distinct workers each one will process jobs depending on an Enum contained inside the Job Message. 

Thank you

**Comment by geofmureithi** [comment]
Regarding filtering are  possible approaches:
1. Use tower filter. This would filter out specific jobs only processing those meeting the predicate. https://docs.rs/tower/latest/tower/filter/index.html
2. Create your own custom tower layer.

Tho steer jobs in an enum, Use tower steer https://docs.rs/tower/latest/tower/steer/index.html.

That said what you are trying to do may be against apalis's philosophy of one job per worker.
> My goal is to run two distinct workers each one will process jobs depending on an Enum contained inside the Job Message.

why do you need distinct workers? workers are just a future that streams jobs from source and hence a job could be run by any worker. If you could also give an example of the enum that would be great too.

**Comment by AzHicham** [comment]
Hey, 

Well the idea is to have a worker A than can embark an algo A, and another worker B with algo B. 

The incoming message is something like that: 
```
struct RequestPayload {
   algo: Algo (enum Algo::A or Algo::B) 
   input1
   input2
   ....
}
```

Both workers share the same input. This is really useful because for some deployment in prod I don't need the worker A and for other deployment I need both. 
Since Algo A & B are AI model with heavy dependencies & weights, it's convenient to split workers

But you right I think I can also create 2 differents Message/RequestPayload and each worker handle different type of Message instead of embarking the algo enum in the Message

**Comment by geofmureithi** [comment, marked as answer]
You can use generics too
```
RequestPayload<A>
```

**Comment by geofmureithi** [comment]
Also checkout https://lib.rs/crates/either
